docs/DeveloperGuide.md

```diff
@@ -4,30 +4,148 @@
+ - `storage`: Handles the reading and storing of our data.
+
+## Implementation
+### 'Add' feature
+This mechanism is facilitated by `CommandAdd`, which extends `Command`, as well as `ParserAdd`. `ParserAdd` parses the 
+user input into a string of words `parsedInput[]` which `CommandAdd` then stores internally. `CommandAdd` also stores an 
+internal reference to the list of expenses `expenseList` to be added to.
+
+`CommandAdd` implements the following operations:
+ - `CommandAdd#execute()` -- Instantiates the `Expense` object with the parsed inputs and adds it to `expenseList`
+ - `CommandAdd#roundInput(amount)` -- Rounds the expense amount to 2 decimal places. 
+
+`ParserAdd` implements the following operations:
+ - `ParserAdd#parseInput(userInput)` -- Parses the user input into an array of strings for `CommandAdd` to read.
+ - `ParserAdd#checkType(input)` -- Checks the type of the input corresponding to its index.
+ - `ParserAdd#substringIndex(type)` -- Returns the substring index of the start of the user input.
+
+The `CommandAdd#execute()` operation is exposed in the main `Duke` class, while the operations in `ParserAdd` are
+exposed in the `Parser` class.
+
+Given below is an example usage of the feature.
+![](./diagrams/AddFeature.png)
+
+Step 1. The user executes `add amt/24 t/02-02-2012` as the `userInput` to add a new `Expense` into the the list of 
+expenses. `Duke` calls `Parser#extractCommandKeyword(userInput)` to parse the input and determines that the `add` command is called.
+
+Step 2. `Duke` instantiates a new `CommandAdd` and calls `CommandAdd#execute()`, which in turn calls 
+`Parser#extractAddParameters(userInput)` and `ExpenseList#getExpenseList()`.
+
+Step 3. `Parser#extractAddParameters(userInput)` then calls `ParserAdd#parseInput(userInput)` and returns the parsed 
+input as a string of words `parsedInput[]`.
+
+Step 4. `CommandAdd#execute` instantiates a new `Expense` object with the returned `parsedInput[]` and adds it to
+`expenseList`.
+
+### "Delete" feature
+
+This mechanism is facilitated by `CommandDelete`, which extends `Command`. It makes use of the output from `Parser`,
+which takes in the user input as string and returns the index of the expense to be deleted, and the `CommandDelete`
+proceeds to remove the respective expense from the list of expenses stored in `expenseList`.
+
+`CommandDelete` implements the following operation:
+- `CommandDelete#execute()` -- Removes expense based on the index specified by user.
+
+`Parser` implements the following operation:
+- `Parser#extractIndex(userInput)` -- Extracts the index of the expense to be deleted, as specified by the user.
+
+The `CommandDelete#execute()` operation is exposed in the main `Duke` class. The `Parser#extractIndex(userInput)`
+is exposed in the `parser` class.
+
+Given below is an example usage scenario and how the 'delete' mechanism behaves at each step.
+
+![](./diagrams/DeleteFeature.png)
```

Comment:
Consider extending this sequence diagram to also show what you mentioned in step 3 - how an expense is deleted, in order to add value to this diagram.